app/cafe.py

```diff
@@ -0,0 +1,24 @@
+from datetime import date
+from app.errors import (
+    NotVaccinatedError, OutdatedVaccineError, NotWearingMaskError
+)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> None | str:
+        if "vaccine" not in visitor:
+            raise NotVaccinatedError(
+                "You need to be vaccinated to visit the cafe!"
+            )
+        elif visitor["vaccine"]["expiration_date"] < date.today():
+            raise OutdatedVaccineError(
+                "The vaccination has expired!"
+            )

```

Comment:
Bug: You're trying to compare a date with a dictionary here, which will raise a `TypeError`. You should get the 'expiration_date' from the 'vaccine' dictionary first. Make sure you handle the case where 'expiration_date' might not be present in the 'vaccine' dictionary.